### PR TITLE
Add header_type to OTel documentation

### PR DESCRIPTION
**Pull request by nbialostosky**
### Description

This pull request will add the ability to set the `header_type` for the opentelemetry plugin:

https://github.com/Kong/kong/pull/10620

To ensure the documentation remains in sync with the options available for plugins, this adds the `header_type` key to the `config` section showing its something that can be set.

### Testing instructions

Netlify link: https://deploy-preview-5422--kongdocs.netlify.app/

### Checklist 

- [N/A] Review label added 
- [N/A] PR pointed to correct branch -- I don't know when my PR will be merged and when/where it will be released, so pointing to `main` currently

### CODE REVIEW

File: app/_hub/kong-inc/opentelemetry/_index.md [hunk 1]
```diff
@@ -98,6 +98,15 @@ params:
       datatype: number
       description: |
         The timeout, in milliseconds, for reading the response from the OTLP server.
+    - name: header_type
+      required: true
```
Comment (mheap) [inline app/_hub/kong-inc/opentelemetry/_index.md:102]: Is this accurate? I believe there’s a default value based on the other PR

Comment (nbialostosky) [reply app/_hub/kong-inc/opentelemetry/_index.md:102]: I believe so, below this line it specifies the default as `default: preserve`. However, in the schema of the plugin it does note that it's required, so wanted that to match here (it's similar to the Zipkin implementation too, which has these similar values).

However, if it needs to be updated let me know!

**Comment by nbialostosky** [comment]
I believe this should be rebased off `release/gateway-3.3` now!

**Comment by nbialostosky** [comment]
Something happened on the rebase, so closing this out and just started a new PR that was pointed directly at the proper branch: https://github.com/Kong/docs.konghq.com/pull/5431

**Pull request closed without merging**